GCTC-NTGC/gc-digital-talent
# Feature - submit application page api

Resolves #3333 

## Notes
- [x] When clicking submit, the Submit Application mutation is called
- [x] A toast shows whether submission was successful or failed
- [x] If successful, user is navigated to the My Applications page (which shows all their applications with statuses - need to be worked on sep. currently this is a blank page. )


**UPDATE** these will be handled by issue #3873
- [/] **NEW** Create a new test applicant with a completed profile and a completed application. 
- [/] A cypress test for this (coming soon. still stuck on this unfortunately)

## Testing
- Sign in as `applicant@test.com`
- Go to path `en/browse/applications/{poolCandidateId}/submit`. You will need to grab a poolCandidateId with a `userId` that matches the test applicant (`email: applicant@test.com`).
- The submit button may be disabled if the application is not completed yet. If it is disabled it could be for two reasons: the profile is incomplete OR they don't have all the required skills. You may need to manipulate the applicant profile and/or pools data to change that. I made a new criteria to create a new test applicant with a completed profile and a completed application. This should help with testing. 

(Technically they wouldn't be able to make it to step 2 unless the application was complete.).
